File: medium/marbles/marbles.py
#!/usr/bin/python 


def factorial(n):
	if n == 1 or n == 0 :
		return 1
	else :
		x=1
		for i in range ( 2 , n+1 ) :
			x=x*i
		return x 


cases = int( raw_input() )

while cases :
	nums = raw_input().split() 
	
	n = int(nums[0])
	k = int(nums[1])

	n=n-1
	k=k-1
	ans = factorial( n ) / ( factorial(n-k) * factorial(k) )

	print ans 
	cases = cases-1

File: medium/marbles/marbles.cpp
#include<iostream>

using namespace std;

int main()
{
	long long int n,k ;
	int cases;
	cin>>cases;
	long long ans;
	while( cases-- )
	{
		cin>>n>>k;
//		cout<<fact(n-1)/(fact(k-1)*fact(n-k))<<endl;
		n=n-1;
		k=k-1;
		ans = 1 ;
		long long j;
	        j = 2;	
		for(long long i = n-k+1 ; i<= n ; i++)
		{
			ans=ans*i;

			for ( ;ans%j==0&&j<=k;j++)
			{
				ans=ans/j;
			}
		}
		cout<<ans<<endl;
				 	
	}
}

